R/ShinyArchRExports.R

```diff
@@ -0,0 +1,620 @@
+  .logThis(mget(names(formals()),sys.frame(sys.nframe())), "exportShinyArchR Input-Parameters", logFile = logFile)
+  
+  .requirePackage("shiny", installInfo = 'install.packages("shiny")')
+  .requirePackage("rhandsontable", installInfo = 'install.packages("rhandsontable")')
+  
+  if(length(groupBy) > 1){
+    stop("Only one value is allowed for groupBy.")
+  }
+  
+  if(is.null(cellColEmbeddings)){
+    stop("The cellColEmbeddings parameter must be defined! Please see function input definitions.")
+  } 
+  if(!all(cellColEmbeddings %in% colnames(ArchRProj@cellColData))){
+    stop("Not all entries in cellColEmbeddings exist in the cellColData of your ArchRProj. Please check provided inputs.")
+  }
+  
+  # Check that the embedding exists in ArchRProj@embeddings
+  if(embedding %ni% names(ArchRProj@embeddings)){
+    stop("embedding doesn't exist in ArchRProj@embeddings")
+  }
+  
+  #check that groupBy column exists and doesn't have NA values
+  if (groupBy %ni% colnames(ArchRProj@cellColData)) {
+    stop("groupBy is not part of cellColData")
+  } else if ((any(is.na(paste0("ArchRProj$", groupBy))))) {
+    stop("Some entries in the column indicated by groupBy have NA values. Please subset your project using subsetArchRProject() to only contain cells with values for groupBy")
+  }
+  
+  # get directories paths
+  projDir <- getOutputDirectory(ArchRProj)
+  mainOutputDir <- file.path(projDir, mainDir)
+  subOutputDir <- file.path(projDir, mainDir, subOutDir)
+  supportedMatrices <- c("GeneScoreMatrix", "GeneIntegrationMatrix", "MotifMatrix") #only these matrices are currently supported for ShinyArchR
+  
+  
+  # Make directory for Shiny App 
+  if(!dir.exists(mainOutputDir)) {
+    
+    dir.create(mainOutputDir, showWarnings = TRUE)
+    
+    ## Check the links for the files
+    filesUrl <- data.frame(
+      fileUrl = c(
+        "https://files.corces.gladstone.org/Users/rcorces/ArchR/Shiny/1.0.3/app.R",
+        "https://files.corces.gladstone.org/Users/rcorces/ArchR/Shiny/1.0.3/global.R",
+        "https://files.corces.gladstone.org/Users/rcorces/ArchR/Shiny/1.0.3/server.R",
+        "https://files.corces.gladstone.org/Users/rcorces/ArchR/Shiny/1.0.3/ui.R"
+      ),
+      md5sum = c(
+        "6453814565316d26a9c83bddebaf41d8",
+        "a07b98a777d374df3639f3c961585a47",
+        "faaf6665647e32e44f62320822868872",
+        "b34874b7d130dc88b579853e297c7e88"
+      ),
+      stringsAsFactors = FALSE
+    )
+    
+    .downloadFiles(filesUrl = filesUrl, pathDownload = mainOutputDir, threads = threads)
+    
+  }else{
+    message("Using existing Shiny files...")
+  }
+  
+  dir.create(subOutputDir, showWarnings = FALSE)
+  # dir.create(ArchRProjOutputDir, showWarnings = FALSE)
+  
+  
+  # Create a copy of the ArchRProj 
+  ArchRProjShiny <- ArchRProj
+  # Add metadata to ArchRProjShiny
+  ArchRProjShiny@projectMetadata[["groupBy"]] <- groupBy
+  ArchRProjShiny@projectMetadata[["tileSize"]] <- tileSize
+  units <- tryCatch({
```

Comment:
This is still a problematic tryCatch. `colorBy` is not defined yet so this always evaluates to "values". I'm not sure what the intended purpose of this is. The units appear to be utilized in plotEmbedding to create the legend label

Reply:
this was originally getting units from the ArchRProj (not ArchRProjShiny). Changed it back in [84af7ad](https://github.com/paupaiz/ArchR/commit/84af7ad367960fbe138beca241c3b934dcb073da)